drivers/sensor/renesas/hs400x/hs400x.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_hs400x
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/crc.h>
+
+#define CRC_POLYNOMIAL 0x1D
+#define CRC_INITIAL 0xFF
+
+LOG_MODULE_REGISTER(HS400X, CONFIG_SENSOR_LOG_LEVEL);
+
+struct hs400x_config {
+	struct i2c_dt_spec bus;
+};
+
+struct hs400x_data {
+	int16_t t_sample;
+	uint16_t rh_sample;
+};
+
+static int hs400x_read_sample(const struct device *dev, uint16_t *t_sample, uint16_t *rh_sample)
+{
+	const struct hs400x_config *cfg = dev->config;
+	uint8_t rx_buf[5];
+	int rc;
+
+	rc = i2c_read_dt(&cfg->bus, rx_buf, sizeof(rx_buf));
+	if (rc < 0) {
+		LOG_ERR("Failed to read data from device.");
+		return rc;
+	}
+
+	*rh_sample = sys_get_be16(rx_buf);
+	*t_sample = sys_get_be16(&rx_buf[2]);
+
+	/*
+	 * The sensor sends a checkum after each measurement. See datasheet "CRC Checksum
+	 * Calculation" section for more details on checking the checksum.
+	 */
+	#if CONFIG_HS400X_CRC
+	uint8_t crc = crc8(rx_buf, 4, CRC_POLYNOMIAL, CRC_INITIAL, 0);
+
+	if (crc != rx_buf[4]) {
+		LOG_ERR("CRC check failed: computed=%u,expected=%u", crc, rx_buf[4]);
+		return crc;
```

Comment:
Returning `crc` seems wrong.

```suggestion
		return -EIO;
```